### PR TITLE
Template to create Backup Vault (preview)

### PR DESCRIPTION
**Pull request by nilaydshah**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Create Backup Vault (preview) Template with default options

### CODE REVIEW

**Comment by nilaydshah** [comment]
Pipeline is failing with - `The subscription is not registered to use namespace 'Microsoft.DataProtection'.` 
This one being new preview RP, we'll have to get it registered to the subscription.
I see that ARM register provider is POST API hence, I can't include this as part of Template. Is there a way if we can get the test subscription registered with the RP namespace and then re-run the test?

**Comment by nilaydshah** [comment]
Also note that the RP is currently available in limited public regions and not available in sovereign cloud. More details are captured in the documentation links.

**Comment by nilaydshah** [comment]
@bmoore-msft  any inputs on how to get this issue fixed?

File: 101-backup-vault-basic/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vaultName": {
+      "type": "string",
+      "metadata": {
```
Comment (bmoore-msft) [inline 101-backup-vault-basic/azuredeploy.json:7]: does this need to be globally unique?  If so, add that to the description and provide a defaultValue using uniqueString()

File: 101-backup-vault-basic/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+  },
+  "resources": [
+    {
+      "type": "Microsoft.DataProtection/BackupVaults",
+      "apiVersion": "2020-01-01-alpha",
+      "name": "[parameters('vaultName')]",
+      "location": "[parameters('location')]",
+      "identity": {
+        "type": "systemAssigned"
```
Comment (bmoore-msft) [inline 101-backup-vault-basic/azuredeploy.json:38]: how is this used?

File: 101-backup-vault-basic/azuredeploy.parameters.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "vaultName": {
```
Comment (bmoore-msft) [inline 101-backup-vault-basic/azuredeploy.parameters.json:5]: remove this when you add the defaultValue

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: 101-backup-vault-basic/azuredeploy.json (hunk 2 above)
Comment (nilaydshah) [reply 101-backup-vault-basic/azuredeploy.json:38]: to configure backup to any workload, Vault's managed identity needs to be granted necessary permissions on the workload. For that identity must be enabled at first place.

File: 101-backup-vault-basic/azuredeploy.json (hunk 1 above)
Comment (nilaydshah) [reply 101-backup-vault-basic/azuredeploy.json:7]: should be unique within resource group and location. will provide appropriate default value in that case.

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.

File: 101-backup-vault-basic/azuredeploy.parameters.json (hunk 3 above)
Comment (nilaydshah) [reply 101-backup-vault-basic/azuredeploy.parameters.json:5]: removed this.

File: 101-backup-vault-basic/azuredeploy.json (hunk 1 above)
Comment (nilaydshah) [reply 101-backup-vault-basic/azuredeploy.json:7]: added default value.

**Comment by nilaydshah** [comment]
@bmoore-msft  Addressed PR comments. Request to reopen for review.

**Pull request closed without merging**